### DISCUSSION TITLE
Using celery with flask application factory pattern

### DISCUSSION THREAD

**Discussion by capymind** [celery/celery#7985] [Q&A] [answered]
Hello, everyone.
First of all, thanks a lot for maintaining this beautiful and invaluable open source project🎉

I know that there are already many "tutorial" explaining using celery in flask.
**BUT, almost(if not all) just start with `app=Flask(__name__)` , which in production nobody use directly initialization of Flask instance.**

So... what I want to achieve is to interagte flask application factory with celery instance including flask context.

The official flask docs say the below code

```python
from celery import Celery

def make_celery(app):
    celery = Celery(app.import_name)
    celery.conf.update(app.config["CELERY_CONFIG"])

    class ContextTask(celery.Task):
        def __call__(self, *args, **kwargs):
            with app.app_context():
                return self.run(*args, **kwargs)

    celery.Task = ContextTask
    return celery
```

That's great. But the point I want to mention is below

```python
from flask import Flask

/*
====
I want to apply application factory pattern, not just direct creation of instance below
(reference) "https://flask.palletsprojects.com/en/2.2.x/patterns/appfactories/"
====
*/

flask_app = Flask(__name__)  # <=🔴😢  
flask_app.config.update(CELERY_CONFIG={
    'broker_url': 'redis://localhost:6379',
    'result_backend': 'redis://localhost:6379',
})
celery = make_celery(flask_app)

@celery.task()
def add_together(a, b):
    return a + b
```

If flask app is created like
```python
from flask import Flask
def create_app():
    app = Flask()
    return app
```

**How to integrate the celery app with flask app with correct way?**

I know that flask and celery is launched decades ago, 
I guess that ALREADY there are solutions(or best practices) about that in online.
But I cannot find solution (maybe my tech skills and searching ability is below-average... sorry about that)

Thanks in advance.
Best regards.

**Comment by auvipy** [comment, marked as answer]
I believe asking this in flask users forum would be more beneficial and relevant

**Reply by capymind** [reply]
Thanks for quick reply!
Um.. I found wrong place, sorry for that 😓.
